packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -545,6 +615,33 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     this.securityGroups.push(securityGroup);
   }
 
+  /**
+   * Emit all group metrics, these metrics describe the group rather than any of its instances
+   * see all available types https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-instance-monitoring.html#as-group-metrics
+   * if granularity is not provided, 1Minute will be set
+   * @default - disabled
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-metricscollection.html
+   */
+  public emitAllMetricsCollections(options?: MetricsCollectionOptions) {
```

Comment:
Why do this as a separate method instead of doing this in the constructor with `MetricsCollectionOptions` as an optional property in `AutoScalingGroupProps`?

Reply:
That was the approach I suggested in: https://github.com/aws/aws-cdk/pull/6712#pullrequestreview-377316838, but due to the cumbersome nature of this property I have suggested this API, see the comment on the pull request. 
For example, the MetricsCollection property in CF is actually a list of MetricsCollection objects, [CF docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html#cfn-as-group-metricscollection). The thought behind implementing the `emitAllMetricsCollections` and `emitMetricsCollection` was to simplify this by having the API name resembling what the user wants to do - emit the metric collections.

Reply:
I'm not opposed to having a separate method, just that this should be configurable via the constructor. 

> but due to the cumbersome nature of this property

Not clear what you are referring to as cumbersome. `MetricsCollectionOptions` has one property `granularity`.

I couldn't quite follow how your comment in the other PR is related here. Could you re-state? sorry.